### PR TITLE
adds npmignore files for packages

### PR DESCRIPTION
**Pull request by o-t-w**
if there is no .npmignore file, create an .npmignore file

### CODE REVIEW

File: lib/js/_publish.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const error = require('./_utils/_error');
 const extendPackage = require('./_utils/_extend-package').publish;
 const getPackages = require('./_utils/_get-packages');
 
+const npmIgnore = require('./_publish/_npm-ignore');
```
Comment (alexkilgour) [inline lib/js/_publish.js]: to follow the existing naming conventions, this should be `_generate-npm-ignore`

File: lib/js/_publish/_npm-ignore.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+'use strict';
+
+const {promisify} = require('util');
+const fs = require('fs');
+
+const writeFileAsync = promisify(fs.writeFile);
+const accessAsync = promisify(fs.access);
+
+const stuffToIgnore = `__tests__ \n`;
+
```
Comment (alexkilgour) [inline lib/js/_publish/_npm-ignore.js]: `patternsToIgnore` would be more descriptive

File: lib/js/_publish/_npm-ignore.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+function npmIgnore(path) {
+	return accessAsync(`${path}/.npmignore`, fs.constants.F_OK)
+		.catch(() => writeFileAsync(`${path}/.npmignore`, stuffToIgnore));
+}
```
Comment (alexkilgour) [inline lib/js/_publish/_npm-ignore.js]: what if `writeFileAsync` fails?

**Comment by alexkilgour** [comment]
Can you add a test for the new file. There is already methods for mocking `fs` functions

**Comment by jpw** [comment]
We have yet to see a need for this so closing.

**Pull request closed without merging**